### DISCUSSION TITLE
lpf.SetFrequency(15000 / hw.AudioSampleRate());

### DISCUSSION THREAD

**Discussion by IC-Alchemy** [clevelandmusicco/HothouseExamples#7] [General]
In ModDelay  
```

  lpf.SetFrequency(15000 / hw.AudioSampleRate());
        hpf.SetFrequency(20 / hw.AudioSampleRate());
```

I think you meant:
  ```
 lpf.SetFrequency(15000 );
 hpf.SetFrequency(20 ); 
```

**Comment by clevelandmusicco** [comment]
The DaisySP implementation is a bit confusing, but I seem to recall the input to `SetFrequency()` is a ratio of the filter cutoff frequency to the samplerate. This would make sense given in a DSP environment, we can only accurately recreate frequencies up to half the sample rate.

From `onepole.h`:

```cpp
    inline void SetFrequency(float freq)
    {
        // Clip coefficient to about 100.
        freq = freq < 0.497f ? freq : 0.497f;
 
        g_  = tanf(PI_F * freq);
        gi_ = 1.f / (1.f + g_);
    }
```

So, the valid range for `freq` is from 0 to 0.497f, and `lpf.SetFrequency(desired_cutoff/sample_rate)` works as expected, with both values in Hz. I'll confirm in the next few days when I am at a dev workstation, but let me know if I've missed something!

**Reply by clevelandmusicco** [reply]
Confirmed ^^

**Comment by clevelandmusicco** [comment]
Also, I should mention that ModDelay is the only effect in the repo that isn't complete/tested; it's there by accident, actually. Sorry about that!
